### PR TITLE
[218] Develop a List View for Customer Orders And Store The Data in A Cookie

### PR DESCRIPTION
**Pull request by Israel-Force**
## Task: `Develop a List View for Customer Orders And Store The Data in A Cookie`
- `added an order.php file for the second task on php and db`
Screenshot (21)
Screenshot (20)

### CODE REVIEW

File: submissions/israelOladosu/php-db/classes/queryDb.class.php [hunk 1]
```diff
@@ -30,4 +30,20 @@ public function getAllCustomers()
         }
     }
 
-};
\ No newline at end of file
+    public function getOrders($id)
+    {
+        try {
+
+            $sql = 'SELECT orders.id, items.quantity, items.unit_price, items.total_amount, orders.made_at, orders.delivery_address, products.product_name FROM items LEFT JOIN orders ON orders.customer_id = items.order_id LEFT JOIN products ON products.id = items.product_id WHERE orders.customer_id = ?';
```
Comment (adaugochi) [inline submissions/israelOladosu/php-db/classes/queryDb.class.php:37]: kindly break this line of code

Comment (adaugochi) [reply submissions/israelOladosu/php-db/classes/queryDb.class.php:37]: Also, why passing `?` in this statement `WHERE orders.customer_id = ?`

Comment (adaugochi) [reply submissions/israelOladosu/php-db/classes/queryDb.class.php:37]: Also, is this how `orders.customer_id = items.order_id`  relation between `orders` and `items` tables exist? I think it suppose to be `orders.id = items.order_id`.  Then you can get the customer_id on the `orders` table. That is now a relationship between the `orders` and the `addresses` or `customers` table

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/israelOladosu/php-db/classes/queryDb.class.php (hunk 1 above)
Comment (Israel-Force) [reply submissions/israelOladosu/php-db/classes/queryDb.class.php]: > Also, why passing `?` in this statement `WHERE orders.customer_id = ?`

no reason in particular, just wanted to try out the use of placeholders as we were taught in the lesson.

Comment (Israel-Force) [reply submissions/israelOladosu/php-db/classes/queryDb.class.php]: > Also, is this how `orders.customer_id = items.order_id` relation between `orders` and `items` tables exist? I think it suppose to be `orders.id = items.order_id`. Then you can get the customer_id on the `orders` table. That is now a relationship between the `orders` and the `addresses` or `customers` table

Right, Thanks for the correction.

Comment (adaugochi) [reply submissions/israelOladosu/php-db/classes/queryDb.class.php]: > > Also, why passing `?` in this statement `WHERE orders.customer_id = ?`
> 
> no reason in particular, just wanted to try out the use of placeholders as we were taught in the lesson.

ok. You can leave it then

File: submissions/israelOladosu/php-db/classes/queryDb.class.php [hunk 2]
```diff
@@ -30,4 +30,27 @@ public function getAllCustomers()
         }
     }
 
+    public function getOrders($id)
+    {
+        try {
+
+            $sql = "SELECT items.quantity, items.unit_price,
+            items.total_amount, orders.made_at, orders.id, 
+            orders.delivery_address, products.product_name
```
Comment (adaugochi) [inline submissions/israelOladosu/php-db/classes/queryDb.class.php:39]: I don't think there is a column called `delivery_address` on orders table. I think the customer location should be coming from the `addresses` table

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/israelOladosu/php-db/classes/queryDb.class.php (hunk 2 above)
Comment (Israel-Force) [reply submissions/israelOladosu/php-db/classes/queryDb.class.php]: oh! I actually altered the orders table to have a delivery_address column forgetting totally that I had a table for addresses. I have made the necessary adjustment now. I will also add the new screenshots

Comment (Israel-Force) [reply submissions/israelOladosu/php-db/classes/queryDb.class.php]: Added the new screenshot now. Thank you.

**Review by adaugochi — APPROVED** [review]
well done

**Pull request merged**